### PR TITLE
Redis Store input and output connectors

### PR DESCRIPTION
**Pull request by ishark**
It contains following changes:
- Added concrete Input connectors for Redis to get Key-Value pairs and Map Datatypes for given set of keys
- Added an implementation for writing <key, POJO> to Redis as a <key. Map> based on input expressions
- Added a test case for POJO connector.

### CODE REVIEW

File: contrib/src/main/java/com/datatorrent/contrib/common/FieldInfo.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
+package com.datatorrent.contrib.common;
```
Comment (ishark) [inline contrib/src/main/java/com/datatorrent/contrib/common/FieldInfo.java]: This file is taken as is from Bright's puill request.

Comment (pm48) [inline contrib/src/main/java/com/datatorrent/contrib/common/FieldInfo.java]: Please check if this is the right way to specify user configured parameters.We were specifying setter/getter for them directly on operator for other concrete implementations.

Comment (pm48) [inline contrib/src/main/java/com/datatorrent/contrib/common/FieldInfo.java]: SupportType enum can just have different data types as fields in it.Do we need rest of the code in it?

**Comment by pm48** [comment]
You can use Logger from
org.slf4j.Logger;
 org.slf4j.LoggerFactory;
instead of System.out.println in src java files.

**Comment by pm48** [comment]
Is protected List<String> keys set by user? If yes please provide javadoc on top of getter.

**Comment by pm48** [comment]
Please rename ObjectToRedisOutputOperator

File: contrib/src/main/java/com/datatorrent/contrib/redis/RedisKeyValueInputOperator.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * This is the an implementation of a Redis input operator for fetching Key-Value pair stored in Redis.
+ * It takes in keys to fetch and emits corresponding <Key, Value> Pair. Value data type is String in this case.
+ * <p></p>
+ * @displayName Redis Input Operator for Key Value pair
+ * @category Store
+ * @tags input operator, key value
+ *
+ */
+public class RedisKeyValueInputOperator extends  AbstractStoreInputOperator<KeyValPair<String, String>, RedisStore>{
```
Comment (PramodSSImmaneni) [inline contrib/src/main/java/com/datatorrent/contrib/redis/RedisKeyValueInputOperator.java]: Research how to implement a progressive read input operator that fetches new data from the store as it is available

File: contrib/src/main/java/com/datatorrent/contrib/redis/RedisMapAsValueInputOperator.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * This is the an implementation of a Redis input operator
+ * It takes in keys to fetch and emits Values stored as Maps in Redis i.e. when value datatype in Redis is HashMap
+ * <p></p>
+ * @displayName  Redis Input Operator for Map
+ * @category Store
+ * @tags input operator, key value
+ *
+ */
+
+public class RedisMapAsValueInputOperator extends AbstractStoreInputOperator<Map<String, String>, RedisStore> {
```
Comment (PramodSSImmaneni) [inline contrib/src/main/java/com/datatorrent/contrib/redis/RedisMapAsValueInputOperator.java]: Same as above. Should be progressive read.

File: contrib/src/main/java/com/datatorrent/contrib/redis/RedisPOJOOutputOperator.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
+package com.datatorrent.contrib.redis;
```
Comment (PramodSSImmaneni) [inline contrib/src/main/java/com/datatorrent/contrib/redis/RedisPOJOOutputOperator.java]: Copyright

**Pull request closed without merging**